apps/src/applab/gridUtils.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * A couple of utility functions for dealing with our design mode grid.
+ */
+module.exports = {
+  GRID_SIZE: 5,
+  /**
+   * Given an element being dragged, determine the scaled x/y position of the
+   * top left corned, scaled to our visualization.
+   * @param {jQueryObject} draggingElement
+   * @return {object}
+   */
+  scaledDropPoint: function (draggingElement) {
+    var div = document.getElementById('designModeViz');
+
+    var xScale = div.getBoundingClientRect().width / div.offsetWidth;
+    var yScale = div.getBoundingClientRect().height / div.offsetHeight;
+
+    var left = (draggingElement.offset().left - $(div).offset().left) / xScale;
+    var top = (draggingElement.offset().top - $(div).offset().top) / yScale;
+
+    // snap top-left corner to nearest location in the grid
+    left = this.snapToGridSize(left);
+    top = this.snapToGridSize(top);
```

Comment:
I'm going to push back a little bit with the following statements
(a) What I have now is currently working and reasonably understandable (I think)
(b) We have a lot of Applab bugs we still want to fix before exiting beta, and time spent refactoring this further is time spent not fixing other bugs.

I think you're absolutely right that we could make this code cleaner still, and I think you have some good thoughts on how to do it (and I also very much appreciate the feedback). That said, given how much other stuff we have, I claim this is "good enough" for the time being. Thoughts?